Public/Src/Tools/SandboxedProcessExecutor/Executor.cs

```diff
@@ -0,0 +1,322 @@
+
+            if (executeResult.result != null)
+            {
+                if (!TryWriteSandboxedProcessResult(executeResult.result))
+                {
+                    return ExitCode.FailedWriteOutput;
+                }
+            }
+
+            return executeResult.exitCode;
+        }
+
+        private bool TryReadSandboxedProcessInfo(out SandboxedProcessInfo sandboxedProcessInfo)
+        {
+            sandboxedProcessInfo = null;
+
+            try
+            {
+                using (FileStream stream = File.OpenRead(Path.GetFullPath(m_configuration.SandboxedProcessInfoInputFile)))
+                {
+                    // TODO: Custom DetoursEventListener?
+                    sandboxedProcessInfo = SandboxedProcessInfo.Deserialize(stream, m_loggingContext, detoursEventListener: null);
+                }
+            }
+            catch (IOException ioException)
+            {
+                ConsoleError(ioException.ToString());
+                return false;
+            }
+
+            return true;
+        }
+
+        private bool TryWriteSandboxedProcessResult(SandboxedProcessResult result)
+        {
+            Contract.Requires(result != null);
+
+            try
+            {
+                using (FileStream stream = File.OpenWrite(Path.GetFullPath(m_configuration.SandboxedProcessResultOutputFile)))
+                {
+                    result.Serialize(stream);
+                }
+            }
+            catch (IOException ioException)
+            {
+                ConsoleError(ioException.ToString());
+                return false;
+            }
+
+            return true;
+        }
+
+        private bool TryPrepareSandboxedProcess(SandboxedProcessInfo info)
+        {
+            Contract.Requires(info != null);
+
+            FileAccessManifest fam = info.FileAccessManifest;
+
+            if (!string.IsNullOrEmpty(fam.InternalDetoursErrorNotificationFile))
+            {
+                Analysis.IgnoreResult(FileUtilities.TryDeleteFile(fam.InternalDetoursErrorNotificationFile));
+            }
+
+            if (fam.CheckDetoursMessageCount && !OperatingSystemHelper.IsUnixOS)
+            {
+                string semaphoreName = fam.InternalDetoursErrorNotificationFile.Replace('\\', '_');
+
+                if (!fam.SetMessageCountSemaphore(semaphoreName))
+                {
+                    ConsoleError($"Semaphore '{semaphoreName}' for counting Detours messages is already opened");
+                    return false;
+                }
+            }
+
+            if (info.GetCommandLine().Length > SandboxedProcessInfo.MaxCommandLineLength)
+            {
+                ConsoleError($"Process command line is longer than {SandboxedProcessInfo.MaxCommandLineLength} characters: {info.GetCommandLine().Length}");
+                return false;
+            }
+
+            m_outputErrorObserver = OutputErrorObserver.Create(info);

```

Comment:
>OutputErrorObserver [](start = 36, length = 19)

Do we need to be processing output differently for SandboxedProcessExecutor? Can we just pipe it directly to SandboxedProcessPipExecutor and just leverage the same code that's defined there?

Reply:
This is what I'm doing. This observer is essentially used to observe warning lines, and increment the counter. We do the same thing in  `SanboxedProcessPipExecutor.cs`. If we want to use the code in `SanboxedProcessPipExecutor.cs`, then I need to establish a pipe between the sandboxed executor tool and `SandboxedProcessPipExecutor.cs`. Currently, we don't want to go into that direction because establishing pipes leads us to implementing client/server, which is a bit error prone, and moreover, we need to understand the security policy for those pipes in VM.

---
In reply to: [276486637](https://github.com/Microsoft/BuildXL/pull/159#discussion_r276486637) [](ancestors = 276486637)